### ISSUE TITLE
Should dracut uncompress compressed kernel modules before installing into the initramfs?

### ISSUE THREAD

**Issue by Nowa-Ammerlaan** [dracutdevs/dracut#2669]
A kernel that is configured to compress modules (for example `CONFIG_MODULE_COMPRESS_XZ=y`) still supports loading uncompressed modules. Given that it is size-wise more efficient to compress a bunch of uncompressed files, compared to compressing a bunch of files that are individually already compressed. It might make sense for Dracut to decompress modules it finds before including them in the initramfs.

The same holds for firmware files (`CONFIG_FW_LOADER_COMPRESS_XZ=y`).

To put this in numbers, an UKI for the same kernel with the same user space utilities, but with pre-xz-compressed kernel modules and firmware is 356.352 kb larger on my system.

**Comment by Artoria2e5** [comment]
Yes, yes, yes. This should absolutely be an option: I did some testing in March 2024 and it helps a strong initramfs compressor (e.g. zstd -15) compress quite a bit further. Compressed firmware is a candidate too. https://github.com/AOSC-Dev/aosc-os-abbs/discussions/6006#discussioncomment-9443065

patch https://github.com/dracut-ng/dracut-ng/pull/1176

**Comment by Artoria2e5** [comment]
Stale bot considered harmful, chapter 1048577

well it got me up to clean up a PR. maybe it helps
